lib/ood_core/job/adapters/kubernetes/resources.rb

```diff
@@ -33,13 +33,47 @@ def initialize(data)
     end
   end
 
+  class Probe
+    attr_accessor :initial_delay_seconds, :failure_threshold, :period_seconds
+
+    def initialize(data)
+      data ||= {}
+      @initial_delay_seconds = data[:initial_delay_seconds] || 2
+      @failure_threshold = data[:failure_threshold] || 5
+      @period_seconds = data[:period_seconds] || 5
+    end
+
+    def to_h
+      {
+        initial_delay_seconds: initial_delay_seconds,
+        failure_threshold: failure_threshold,
+        period_seconds: period_seconds,
+      }
+    end
+  end
+
+  class TCPProbe < Probe
```

Comment:
I don't think we need the subclass. I think we can just start with 1 class and modify later if required. But you're right about tcp being better than http.

Reply:
I merged `Probe` into `TCPProbe` so now there is only 1 class.